index.html

```diff
@@ -1,453 +1,582 @@
+        <code>EIP712Domain</code> property that contains the JSON schema according to the EIP712
+        specification. Remaining entries MUST be the JSON schemas of the message to be
+        signed in the EIP712 format. <code>types</code> MUST contain a property <code>proof</code> of type
+        <code>Proof</code>
+        which contains the JSON schema for all properties of the final linked data signature <code>proof</code> property
+        that need to be signed.
+      </li>
+      <li>
+        <code>message</code> MUST be the linked data object that contains the message to be signed. The message MUST
+        contain a <code>proof</code> property with values set to the values of
+        the properties in the resulting linked data signature <code>proof</code> property that are expected to be
+        signed.
+      </li>
+      <li>
+        <code>domain</code> MUST have the value as defined in <code>EIP712</code>.
+      </li>
+      <li>
+        <code>primaryType</code> MUST have the value as defined in <code>EIP712</code>. <code>primaryType</code>
+        represents the top-level type of the object in the EIP712 <code>message</code> but does not have to correspond
+        to any of
+        the types in the <code>message</code>.
+      </li>
+    </ul>
 
     <section>
-      <h2>Introduction</h2>
+      <h2>Types Generation</h2>
       <p>
-        This specification defines a cryptographic suite for the purpose of
-        creating, verifying proofs for EIP712 signatures in conformance
-        with the Linked Data Proofs [[LD-PROOFS]] specification.
-      </p>
-      <p>
-        In general the suites uses the JCS Algorithm
-        [[JCS]] to transform an input document into its
-        canonical form. The cannonical representation is then provided 
-        to the EIP712 signature function.
+        If <code>TypedData</code> is not provided to the signature suite, the suite MUST generate the JSON object by
+        inferring types from the input document, and optionally a provided `primaryType`.
       </p>
 
       <p>
-        A signature scheme consists of a hashing algorithm and a signing algorithm.
-        The signing algorithm of choice in Ethereum is secp256k1. The hashing algorithm of
-        choice is keccak256.
+        In case of ambiguous types, the algorithm SHOULD defer to using the most liberal option. For example, a number
+        should be inferred as the <code>uint256</code> type even if that specific number can fit in a <code>uint8</code>
+        type.
       </p>
 
       <p>
-        EIP712 defines a standard API for Web3 Provider (e.g., wallets) to generate signatures
-        over human-readable data where the signature can be verified either by a Smart Contract
-        on the Ethereum Blockchain, or completely offchain. 
-      </p>
+        The <a>types generation algorithm</a> is defined as follows:
+      <ul>
```

Comment:
To make it appear with numbered steps as in other algorithms (e.g. [Create Verify Data Algorithm in BBS+ Signature Suite 2020](https://w3c-ccg.github.io/ldp-bbs2020/#create-verify-data-algorithm-0), change the `ul` to `ol` and add 
class "algorithm".
Also, the list should be moved outside the paragraph (p) tag, for HTML correctness.
```diff
@@ -403,3 +403,4 @@
         The <a>types generation algorithm</a> is defined as follows:
-      <ul>
+      </p>
+      <ol class="algorithm">
         <li>Creates a mapping <code>output</code> from <code>string</code> to <code>TypedDataField[]</code> types, where
@@ -418,3 +419,3 @@
           document, and <code>type</code> set to the corresponding EIP712 primitive type
-          <ul>
+          <ol class="algorithm">
             <li><code>boolean</code> - maps to <code>bool</code></li>
@@ -422,3 +423,3 @@
             <li><code>string</code> - maps to <code>string</code></li>
-          </ul>
+          </ol>
         </li>
@@ -427,3 +428,3 @@
           <code>type</code> set to the corresponding EIP712 array type
-          <ul>
+          <ol class="algorithm">
             <li><code>boolean[]</code> - maps to <code>bool[]</code></li>
@@ -431,3 +432,3 @@
             <li><code>string[]</code> - maps to <code>string[]</code></li>
-          </ul>
+          </ol>
           <p class="example" title="Arrays of structs are not yet supported">
@@ -441,3 +442,3 @@
           value equal to <code>_recursiveOutput</code>.
-          <ul>
+          <ol class="algorithm">
 
@@ -453,3 +454,3 @@
               add them directly to <code>output</code></li>
-          </ul>
+          </ol>
         </li>
@@ -459,4 +460,3 @@
         </li>
-      </ul>
-      </p>
+      </ol>
```